### PR TITLE
Test coverage for node and cluster listing

### PR DESCRIPTION
**Pull request by debloper**
- Mock data taken from live server
- Fixes Jira ticket TEN-152 and TEN-153

Signed-off-by: Soumya Deb <debloper@gmail.com>

### CODE REVIEW

**Comment by debloper** [comment]
@anivargi @gnehapk @kamleshverma1 ping for review.

File: test/unit/cluster-list-test.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+describe("Unit: List Clusters", function () {
```
Comment (kamleshverma1) [inline test/unit/cluster-list-test.js]: We should not mention specifically name as "list" in file name as well in describe block . File name should be "cluster-controller-test.js" and inside describe should be "UNIT CONTROLLER - clusterController".

File: test/unit/cluster-list-test.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        module("TendrlModule");
+        module("TestDataModule");
+        module("templates");
+    });
+
+    beforeEach(function () {
+
+        var templateHtml;
+
+        inject(function (_$q_, _$controller_, _$rootScope_, _$state_, _$templateCache_, _$compile_, _clusterList_, _$location_) {
```
Comment (kamleshverma1) [inline test/unit/cluster-list-test.js]: _$location is not getting used . we can remove this.

File: test/unit/node-list-test.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
+describe("Unit: List Nodes", function () {
```
Comment (kamleshverma1) [inline test/unit/node-list-test.js]: same as above comment .

File: test/unit/node-list-test.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        module("TendrlModule");
+        module("TestDataModule");
+        module("templates");
+    });
+
+    beforeEach(function () {
+
+        var templateHtml;
+
+        inject(function (_$q_, _$controller_, _$rootScope_, _$state_, _$templateCache_, _$compile_, _nodeList_, _$location_) {
```
Comment (kamleshverma1) [inline test/unit/node-list-test.js]: _$location is not getting used . we can remove this.

**Comment by gnehapk** [comment]
@debloper Can you please check why this build is failing.

**Comment by debloper** [comment]
Tests are failing because:

```
FAILED TESTS:

  Unit Controller: clusterController
    ✖ "before each" hook for "Should get list of clusters"
      PhantomJS 2.1.1 (Linux 0.0.0)
    Can't find variable: _$location_
    test/unit/cluster-controller-test.js:28:36
    ...

  Unit Controller: nodeController
    ✖ "before each" hook for "Should get list of nodes"
      PhantomJS 2.1.1 (Linux 0.0.0)
    Can't find variable: _$location_
    test/unit/node-controller-test.js:28:36
    ...
```

Which means, my initial patch was correct, and removing $location by @kamleshverma1's suggestion broke the tests. I'll revert it.

**Comment by debloper** [comment]
@gnehapk @kamleshverma1 I hope it's alright now.

**Comment by gnehapk** [comment]
Added the unit tests for both list with the updated response to repositry. Hence closing this issue.

**Pull request closed without merging**